### PR TITLE
web-wallet: Restrict mnemonic step input to alphabetical characters (Restore Flow)

### LINKED ISSUE #2355: web-wallet: Restrict Mnemonic Step Input to Alphabetical Characters Only in Restore Flow
### Description:
Currently, in the Mnemonic step of the Restore flow, the user is required to enter words from the mnemonic phrase one by one. However, the input fields allow all types of characters, including spaces and special characters.

### Expected Behavior:
- The input fields in the Mnemonic step should only accept alphabetical characters (A-Z, a-z).
- Any other characters, including spaces, numbers, or special characters, should be disallowed.

### Steps to Reproduce:
1. Go to the Restore flow.
2. Enter a word in the Mnemonic step's input field.
3. Notice that the field accepts non-alphabetical characters, such as numbers, spaces, and special characters.

### Proposed Solution:
- Implement input validation that restricts the input to alphabetical characters only.
- Display an error message or prevent entry when non-alphabetical characters are detected.

### Additional Context:
Restricting input to only alphabetical characters will improve the user experience and reduce potential errors during the mnemonic restore process.

### PR DESCRIPTION
**Pull request by nortonandreev**
Resolves #2355

### CODE REVIEW

File: web-wallet/src/lib/dusk/components/Mnemonic/Mnemonic.svelte [hunk 1]
```diff
@@ -57,11 +57,25 @@
...
+   * Prevents non-alphabetical characters from being entered
+   * and auto-selects the first suggestion on Enter
+   * if there is only one word available
+   * @param {KeyboardEvent} event
    * @param {string} index
    */
   function handleKeyDownOnAuthenticateTextbox(event, index) {
+    const isAlphabetical = /^[a-zA-Z]+$/;
+
+    if (!isAlphabetical.test(event.key)) {
```
Comment (ascartabelli) [inline web-wallet/src/lib/dusk/components/Mnemonic/Mnemonic.svelte:69]: Shouldn't this check be after the "enter key" one?

The second one won't fire for the last suggestion as the enter key is not an alphabetical char.

**Review by ascartabelli — COMMENTED** [review]
Check the behaviour and please if you have time add a test for the new check.

File: web-wallet/src/lib/dusk/components/Mnemonic/Mnemonic.svelte (hunk 1 above)
Comment (nortonandreev) [reply web-wallet/src/lib/dusk/components/Mnemonic/Mnemonic.svelte:69]: `event.key` is `Backspace` for the backspace key and `Enter` for the enter – checked against the `isAlphabetical` regex, both will go through.

**Review by ascartabelli — DISMISSED** [review]

**Review by ascartabelli — APPROVED** [review]

**Comment by nortonandreev** [comment]
Have created an issue to expand the tests: https://github.com/dusk-network/rusk/issues/2355

**Pull request merged**